app.js

```diff
@@ -11,10 +11,57 @@ var express = require('express'),
 Habitat.load();
 
 var app = express(),
-  env = new Habitat();
+  env = new Habitat(),
+  OPTIMIZELY_ID = env.get('OPTIMIZELY_ID'),
+  OptimizelyClient = require('optimizely-node-client'),
+  optimizely = new OptimizelyClient(env.get('OPTIMIZELY_TOKEN'));
 
 app.set('trust proxy', true);
 
+app.use( '/optimizely-experiments.js', function( req, res ) {
+  optimizely.getExperiments(OPTIMIZELY_ID).then(function(experiments) {
+    var experimentPromises = [];
+    var experimentData = {};
+    experimentPromises.push(new Promise(function(resolve) {
+      var variationsPromises = [];
+      experiments.forEach(function(experiment) {
+        experimentData[experiment.id] = {};
+        experiment.variation_ids.forEach(function(variationId) {
+          variationsPromises.push(optimizely.getVariation({id: variationId}).then(function(variation) {
+            if (variation.js_component) {
+              experimentData[experiment.id][variation.id] = variation.js_component;
+            }
+          }));
+        });
+      });
+
+      Promise.all(variationsPromises).then(resolve);
+    }));
+
+    function insertComma(index, arr) {
+      if (index < arr.length-1) {
+        return ","
+      }
+      return "";
+    }
+
+    Promise.all(experimentPromises).then(function() {
+      var jsComponents = "window.optimizelyExperimentData = {\n";
+      Object.keys(experimentData).forEach(function(experiment, index, arr) {
+        jsComponents += "  \"" + experiment + "\": {\n";
+        Object.keys(experimentData[experiment]).forEach(function(variation, index, arr) {
+          jsComponents += "    \"" + variation + "\": function() {\n";
```

Comment:
lol seems github is having a hard time with syntax highlighting for escaped double quotes.